### PR TITLE
fix: add link library boost_system

### LINKED ISSUE #513: Bug: hybridse compile error
missing link library libboost_system

/usr/bin/ld: CMakeFiles/ast_node_converter_test.dir/planv2/ast_node_converter_test.cc.o: undefined reference to symbol '_ZN5boost6system16generic_categoryEv'

os: ubuntu18.04
gcc version: 8.4.0

### PR DESCRIPTION
**Pull request by zoyopei**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Issues: #513 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Review by jingchen2222 — COMMENTED** [review]
We had better add boost_system lib in OpenMLDB/CMakeList.txt as well.

```cmake
set(VM_LIBS hybridse_core hybridse_sdk hybridse_flags boost_filesystem boost_date_time boost_regex)
```

**Comment by jingchen2222** [comment]
Should we add compile test for ubuntu os @aceforeverd . Can run on `main push` event.

**Comment by aceforeverd** [comment]
> Should we add compile test for ubuntu os @aceforeverd . Can run on `main push` event.
> 
> 

Maybe not. Free job number is limited.

**Comment by aceforeverd** [comment]
Actually I found there are many build error on ubuntu 18.04.  see https://github.com/aceforeverd/fedb/runs/3996131455?check_suite_focus=true
- the `find(GTest)` need extra line to set `GTEST_ROOT` to `${CMAKE_PREFIX_PATH}/lib64`
- many link errors. looks like thirdparty asserts is not portable

File: hybridse/CMakeLists.txt [hunk 1]
```diff
@@ -88,10 +88,10 @@ find_package(Threads)
 
 set(Boost_NO_BOOST_CMAKE ON)
 set(BOOST_ROOT "${CMAKE_PREFIX_PATH}")
-find_package(Boost COMPONENTS filesystem date_time regex REQUIRED)
+find_package(Boost REQUIRED COMPONENTS filesystem date_time regex system)
 if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
     set(OS_LIB ${CMAKE_THREAD_LIBS_INIT} rt)
-    set(COMMON_LIBS boost_date_time boost_filesystem boost_regex leveldb protobuf glog unwind z snappy ssl crypto dl)
+    set(COMMON_LIBS boost_date_time boost_filesystem boost_regex boost_system leveldb protobuf glog unwind z snappy ssl crypto dl)
```
Comment (aceforeverd) [inline hybridse/CMakeLists.txt:94]: is `boost_system` a transitive dependence?  I can't find boost system usage on code base

Comment (vagetablechicken) [reply hybridse/CMakeLists.txt:94]: boost_filesystem always needs boost_system, maybe different boost version will show different dependence.
ref https://cmake.org/cmake/help/v3.12/module/FindBoost.html, 'Implicit dependencies such as **Boost::filesystem requiring Boost::system** will be automatically detected and satisfied, even if system is not specified when using find_package and if Boost::system is not added to target_link_libraries. '
If we use boost::filesystem(the target way), maybe we could miss boost::system.

Comment (aceforeverd) [reply hybridse/CMakeLists.txt:94]: yes, `boost::filesystem` is included

**Comment by aceforeverd** [comment]
Hi @zoyopei 
Build directly from ubuntu is not supported yet, since pre-compiled thirdparty is not compellable on ubuntu. We plan to support that in the future. 

It's recommended to build inside our published docker development image instead:
```bash
# from ghcr.io
docker pull ghcr.io/4paradigm/hybridsql:0.4.0
# or dockerhub
docker pull 4pdosc/hybridsql:0.4.0
```

**Pull request closed without merging**